tests/src/Functional/AccessTest.php

```diff
@@ -0,0 +1,153 @@
+    // does not have real support for multiple bundles.
+    BlockContentType::create(['type' => 'group']);
+    Og::groupManager()->addGroup('block_content', 'group');
+
+    // Create a group.
+    $this->group = BlockContent::create([
+      'title' => $this->randomString(),
+      'type' => 'group',
+      'uid' => $this->users['owner']->id(),
+    ]);
+    $this->group->save();
+
+    // Create a group content type.
+    $this->createContentType(['type' => 'group_content']);
+    $settings = [
+      'field_storage_config' => [
+        'settings' => [
+          'target_type' => 'block_content',
+        ],
+      ],
+    ];
+    Og::createField(OgGroupAudienceHelper::DEFAULT_FIELD, 'node', 'group_content', $settings);
+
+    // Grant members permission to edit their own content.
+    /** @var \Drupal\og\Entity\OgRole $role */
+    $role = $this->container->get('entity_type.manager')
+      ->getStorage('og_role')
+      ->load('block_content-group-member');
+    $role->grantPermission('edit own group_content content');
+    $role->save();
+
+    // Subscribe the member to the group.
+    /** @var \Drupal\og\Entity\OgMembership $membership */
+    $membership = OgMembership::create(['type' => OgMembershipInterface::TYPE_DEFAULT]);
+    $membership
+      ->setUser($this->users['member']->id())
+      ->setEntityId($this->group->id())
+      ->setGroupEntityType($this->group->getEntityTypeId())
+      ->addRole($role->id())
+      ->save();
+
+    // Create two group content items, one owned by the group owner, and one by
+    // the member.
+    foreach (['owner', 'member'] as $user) {
+      $this->groupContent[$user] = Node::create([
+        'title' => $this->randomString(),
+        'type' => 'group_content',
+        'uid' => $this->users[$user]->id(),
+        OgGroupAudienceHelper::DEFAULT_FIELD => [['target_id' => $this->group->id()]],
+      ]);
+      $this->groupContent[$user]->save();
+    }
+  }
+
+  /**
+   * Tests access to entity operations through the user interface.
+   *
+   * Note that this does not use a dataProvider because of the slow performance
+   * of setting up functional tests.
+   */
+  public function testEntityOperationAccess() {
+    $test_matrix = [
+      // The administrator should have the right to edit both group content
+      // items.
+      'group-admin' => ['owner' => TRUE, 'member' => TRUE],
+      // The member should only have the right to edit their own group content.
+      'member' => ['owner' => FALSE, 'member' => TRUE],
+      // The non-member cannot edit any group content.
+      'non-member' => ['owner' => FALSE, 'member' => FALSE],
+    ];
+    foreach ($test_matrix as $user => $expected_results) {
+      $this->drupalLogin($this->users[$user]);
+      foreach ($expected_results as $group_content => $expected_result) {
+        $this->drupalGet($this->groupContent[$group_content]->toUrl('edit-form'));
+        $expected_response = $expected_result ? 200 : 403;
```

Comment:
I think we should go with Kernel test. I think this is the only "browser" test, which we could convert to an  API check to the entity. For me it confirms that from the API level we are getting the access right.